### ISSUE TITLE
Discuss including examples

### ISSUE THREAD

**Issue by NathanWalker** [mgechev/angular2-seed#308]
### Having examples in repo

**Pros**:
- You have a lot to go on straight out of the box.
- Less confusing as you don't have to figure out how to integrate example into the particular build setup as they would be made to work out of the box 

**Cons**:
- you have to keep them up to date

**Comment by NathanWalker** [comment]
from @mgechev 

> Currently we're aiming to provide a stable statically typed production and dev gulp builds with the best dev experience possible (we're working on integration with the hot loader). @robwormald, it'd be awesome to discuss your suggestions! Does adding a simple example with the Http module which requests a static resource in the `app` directory seems like a good idea to you?

> Together with @ludohenin tried to collaborate with @juliemr for her [ng2-test-seed](https://github.com/angular/ng2-test-seed) but the goals of both projects are quite different so the Angular 2 seed wasn't able to get in use in this case. @robwormald we'd love to do the same with [angular2-seed](https://github.com/angular/angular2-seed) if you see a bigger intersection between the goals of both projects!

**Comment by alvipeo** [comment]
Let me say this - without these examples (like this repo) I'd be having a hard time to even start. So thanks for them and keeping them up to date.

**Comment by NathanWalker** [comment]
During alpha, it was hard to keep up with all the breaking changes for sure, however now that it's reached `beta`, keeping things up to date should seemingly be smoother sailing.
So maybe the `con` above becomes less of point.

**Comment by NathanWalker** [comment]
From @robwormald
> ...getting some Http ... up in here would be useful. Also usage of the RxJS operators. ...

At a minimum, I agree these would be most useful
- Http
- How to import RxJS operators (since those must be included separately and it's not intuitive how one does this from the start at moment)

**Comment by robwormald** [comment]
I have an example http service and spec and a related component I'll open up a PR for this evening for discussion.

**Comment by ludohenin** [comment]
It's good to have the example in the seed, we should also keep them very easy to remove. Cleaning up many file before starting a new project is can be a pain IMO.
@robwormald excellent:)

**Comment by lihaibh** [comment]
But the seed itself is the example of a project in Angular 2 it just give you the base utilities and structure you need to get started with your project, thats the purpose of seed.. 
Unlike library which you need to have some examples to see how to implement in your project.

**Comment by sclausen** [comment]
The consequence of what @lihaibh wrote could be a fork of angular2-seed packed with a bunch of examples how to implement different stuff like the hot loader, lazy loading or how to use http in different ways. I really like this idea :) maybe accessible by a web frontend showing the example in action.

**Comment by mgechev** [comment]
Externalizing the examples sounds like a good idea. The hot loader belongs to the seed since it is an essential tool which enhances the development experience so we can keep it here.

**Comment by NathanWalker** [comment]
At this point, having 1 `single` external examples repo sounds like the way to go.

I would suggest all the repos here: https://github.com/AngularShowcase
be merged into one examples repo that is a fork of this main seed. That way the project setup can be easily updated with latest changes here and the examples can be easily and independently bootstrapped by modifying `tools/config.ts`.

I can take on the merging of all the repos on `AngularShowcase` into one.
@mgechev or @ludohenin If you want to create a single repo on `AngularShowcase` (maybe simply called `angular2-seed-examples`, I will fork that and consolidate the examples together. After that we could remove the other repos in the showcase, leaving only the one containing all examples together.

This strategy will be similar to what AngularClass did here: https://github.com/AngularClass/angular2-examples

Options:

1. Make the examples repo a fork of this seed. That way the examples repo can be independently run and built right out of the box.

2. Make the example repo just simply examples (like AngularClass) and one would need to place the examples into a `angular2-seed` clone in order to build and try out.

I'm more for option 1 and I can help keep the examples up-to-date as well as keeping the project setup up to date with latest upstream changes from this seed.

**Comment by sclausen** [comment]
I really like option 1

**Comment by mgechev** [comment]
I suggest to use AngularShowcase. Let me know who is interested into adding him/her into the organization where we can collaborate on this.

**Comment by justindujardin** [comment]
I forked this project for a ng2-material example. Nice work, it's a very slick setup.

https://github.com/justindujardin/angular2-seed

**Comment by mgechev** [comment]
Would you add a link to your fork in the README.md?

**Comment by Bigous** [comment]
The discussion is closed, but I'd like suggest to rave an examples directory with linked repositories here to have an index of good examples made in one place (or at least an EXAMPLES.MD to achieve this goal)
What do you think?

**Comment by justindujardin** [comment]
@mgechev I opened #340 as you requested.

**Comment by sclausen** [comment]
@Bigous there is AngularShowcase/angular2-sample-app

**Comment by mgechev** [comment]
Awesome! Thanks!

**Comment by NathanWalker** [comment]
@mgechev If you could add me to https://github.com/AngularShowcase I can help organize things.

@justindujardin nice work!  I vote we add your example to AngularShowcase as either a repo in Showcase or part of a consolidated examples repo (similar to https://github.com/AngularClass/angular2-examples)

Either way, as @mgechev mentioned, using AngularShowcase org for this would be great :+1: 

IMO, it would be excellent to see any and all examples that use this seed on https://github.com/AngularShowcase :)

**Comment by Bigous** [comment]
Hi people, I've made another fork (I don't know why it is not showing at github as a fork, but it is - I'm used to work with svn and I'm learning git, so I might have done something wrong) to integrate this seed with [Highcharts](http://www.highcharts.com).
It's at [Bigous/Ng2Highcharts](https://github.com/Bigous/Ng2Highcharts).
How could I put it at AngularShowcase?

**Comment by NathanWalker** [comment]
Thanks @Bigous I can help you with that. 
I will add your project to AngularShowcase for now.

We may in future consolidate 1 off examples like this into a one repo together. But it will be nice to have your example on the showcase for now. You of course will receive all the credit, nice work! :)

**Comment by NathanWalker** [comment]
@Bigous your Highcharts example implementation is now on the showcase here:
https://github.com/AngularShowcase/angular2-seed-ng2-highcharts

You may want to fork that one and continue to work off the showcase fork to keep any future contributions/changes on the showcase. :+1:

**Comment by mgechev** [comment]
Great! Here's an example section in [the README.md](https://github.com/mgechev/angular2-seed#examples) which can be extended with the links above.
